### ISSUE TITLE
KeepAlive not working

### ISSUE THREAD

**Issue by kartas39** [dotnet/SqlClient#849]
https://github.com/dotnet/SqlClient/blob/699ae37a2ba193c16f653b13400c990249d94569/src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SNI/SNITcpHandle.cs#L821

This method is not called from anywhere. So I think it just doesn't do his stuff.
Please also pay attention that it is wrapped with if NETCOREAPP31_AND_ABOVE, which is defined as:

```
<PropertyGroup Condition="'$(TargetGroup)' == 'netcoreapp' AND !$(TargetFramework.StartsWith('netcoreapp2.'))">
    <DefineConstants>$(DefineConstants);NETCOREAPP31_AND_ABOVE</DefineConstants>
  </PropertyGroup>
```
which does not include .Net5 I assume if you build with net5.0

**Comment by karinazhou** [comment]
Hi @kartas39 ,

I think you are right. This `SetKeepAliveValues` is not referenced anywhere in the driver. I checked the history of the code and it was initially called in System.Data.SqlClient :
https://github.com/dotnet/corefx/blob/release/3.1/src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNITcpHandle.cs#L227

However, when we move from System.Data.SqlClient to Microsoft.Data.SqlClient on Github, this call has been removed. This `SetKeepAliveValues` method is added back in the driver in [PR#395](https://github.com/dotnet/SqlClient/pull/395) which will support the TCP KeepAlive feature in the future. 

Currently, the driver is not officially built against .NET5.0 but it is compatible when you use it in a .NET5.0 application. For a .net core application, the actual driver's runtime library is from \netcoreapp3.1 folder. The SNITcpHandle is part of the managed SNI implementation which is not used on Windows by default. 

Thanks,
